src/Trains.NET.Rendering/Drawing/ImageCache.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+
+namespace Trains.NET.Rendering.Drawing
+{
+    public class ImageCache : IImageCache
+    {
+        private readonly object _cacheLock = new object();
+
+        private readonly Dictionary<object, IImage> _imageBuffer = new();
+        private readonly Dictionary<object, bool> _dirtySTate = new();
+
+        public void Clear()
+        {
+            foreach (object key in _dirtySTate.Keys.ToArray())
+            {
+                _dirtySTate[key] = true;
+            }
+        }
+
+        public IImage Get(object key)
```

Comment:
Could be nice to have a GetOrAdd here?

Reply:
Yeah, I was thinking about whether this should include some way to specify the Add.. might be an interesting simplification in future maybe. Maybe some way to pipe through the change event/dirty flag and have the cache take care of everything